pkg/operation/botanist/monitoring.go

```diff
@@ -424,6 +429,11 @@ func (b *Botanist) DeploySeedMonitoring(ctx context.Context) error {
 
 // DeploySeedGrafana deploys the grafana charts to the Seed cluster.
 func (b *Botanist) DeploySeedGrafana(ctx context.Context) error {
+	// disable monitoring set if disabled in gardenletConfig
+	if !gardenlethelper.IsMonitoringEnabled(b.Config) {
+		return b.DeleteGrafana(ctx)
```

Comment:
Shouldn't we cleanup the monitoring secrets as well? 
https://github.com/gardener/gardener/blob/3d001a092ee093ee579ac1d30ae38e3cca107c3e/pkg/operation/botanist/monitoring.go#L427-L434

Maybe you can combine the two if's?

Reply:
Hmm, don't we use grafana also for loki and logging?

Reply:
yep @plkokanov but that is unchanged. `monitoring` is really just prometheus metrics as far as we can tell

Reply:
> Hmm, don't we use grafana also for loki and logging?

valid point @plkokanov. We don't need it therefore it was removed but I don't really have a strong opinion about it, whatever seems good for you guys :)

Reply:
cc @nickytd @vlvasilev @Kristian-ZH we should keep `grafana` if only the monitoring component is disabled right?

Reply:
Yes, the logging stack uses Grafana too.
You can inert the additional check in your `if`: `gardenlethelper.IsLokiEnabled()`

Reply:
Added loki to the check
```go
	if !gardenlethelper.IsMonitoringEnabled(b.Config) && !gardenlethelper.IsLokiEnabled(b.Config) {
		return b.DeleteGrafana(ctx)
	}
```